File: python_leaoai/python_leaoai/services/Mapper/MappingExtensions.py
from typing import Type, TypeVar
from .generic_mapper import GenericMapper  # supondo que esteja no mesmo módulo

TSource = TypeVar("TSource")
TDestination = TypeVar("TDestination")

def map_to(source: TSource, destination_type: Type[TDestination]) -> TDestination:
    """
    Função utilitária para mapear um objeto para outro tipo,
    usando GenericMapper internamente.
    """
    mapper = GenericMapper(destination_type)
    return mapper.map(source)

File: Framework.Services/Mapper/MappingExtensions.cs
namespace Framework.Services.Mapper
{
    public static class MappingExtensions
    {
        public static TDestination MapTo<TSource, TDestination>(this TSource source)
            where TDestination : new()
        {
            var mapper = new GenericMapper<TSource, TDestination>();
            return mapper.Map(source);
        }
    }

}
